File: Semester 1 - 2018/SENG1120 - Data Structures/Workshops/Week 7/Node.cpp
#include <string>
#include "Node.h"

using namespace std;

Node::Node<valuetype>(value_type newData, Node* newNext, Node* newPrev)
{
	data = newData;
	next = newNext;
	prev = newPrev;
}

Node* Node::get_next()
{
	return next;
}

Node* Node::get_prev()
{
	return prev;
}

value_type Node::get_data()
{
	return data;
}

void Node::set_next(Node* newNext)
{
	next = newNext;
}

void Node::set_prev(Node* newPrev)
{
	prev = newPrev;
}

void Node::set_data(value_type newData)
{
	data = newData;
}

ostream& operator << (ostream& out, Node<value_type>* myAcc)
{
	value_type myData = myAcc->get_data();
	out << "(" << myData.getName() << ", $" << myData.balance() << ")";
	return out;
}

File: Uni/Semester 1 - 2019/SENG2200 - Programming Languages and Paradigms/Assignments/Assignment2/Node.java

/**
 * Defines a Node class, used to traverse an instance of the 
 * MyPolygons linked list
 *
 * Author: Sam Dolbel
 * Date created: 22/3/2019
 * Date modified: 22/3/2019
 */
public class Node
{
    private Node prev;
    private Node next;
    private Polygon data;
    
    public Node(Polygon poly)
    {
        prev = null;
        next = null;
        data = poly;
    }
    
    public Node getNext()
    {
        return next;
    }
    
    public Node getPrev()
    {
        return prev;
    }
    
    public Polygon getData()
    {
        return data;
    }
    
    public void setNext(Node newNext)
    {
        next = newNext;
    }
    
    public void setPrev(Node newPrev)
    {
        prev = newPrev;
    }
    
    public void setData(Polygon newData)
    {
        data = newData;
    }
    
    public void debugPoly()
    {
        data.debugPolygon();
    }
}
